src/components/OptionList/OptionList.js

```diff
@@ -0,0 +1,59 @@
+import React, { PureComponent } from "react";
+import Option from "../Option";
+import "./OptionList.scss";
+
+type Props = {
+  data: Array,
+  type: String
+};
+
+export default class OptionList extends PureComponent<Props, State> {
+  state = {
+    newData: this.props.data
+  };
+  componentDidMount() {
+    this.setState({
+      newData: this.updateItem()
+    });
+  }
+  updateItem() {
+    const { newData } = this.state;
+    return newData.map(item => Object.assign({ active: false }, item));
+  }
+
+  selectItem = item => {
+    const { newData } = this.state;
+    const mappedData = newData.map(mappedItem =>
```

Comment:
While this approach is valid, I'm a little bit concerned that each item is mutated every time the 'selectItem' function is called.